File: CodeHockey/Model/Puck.cs
using System;

namespace Com.CodeGame.CodeHockey2014.DevKit.CSharpCgdk.Model {
    public class Puck : Unit {
        private readonly long ownerHockeyistId;
        private readonly long ownerPlayerId;

        public Puck(long id, double mass, double radius, double x, double y, double speedX, double speedY,
                long ownerHockeyistId, long ownerPlayerId)
                : base(id, mass, radius, x, y, speedX, speedY, 0.0D, 0.0D) {
            this.ownerHockeyistId = ownerHockeyistId;
            this.ownerPlayerId = ownerPlayerId;
        }

        public long OwnerHockeyistId {
            get { return ownerHockeyistId; }
        }

        public long OwnerPlayerId {
            get { return ownerPlayerId; }
        }
    }
}

File: CodeHockey/local_runner/plugins/model/Puck.java
package model;

/**
 * Класс, определяющий хоккейную шайбу. Содержит также все свойства юнита.
 */
public class Puck extends Unit {
    private final long ownerHockeyistId;
    private final long ownerPlayerId;

    public Puck(long id, double mass, double radius, double x, double y, double speedX, double speedY,
                long ownerHockeyistId, long ownerPlayerId) {
        super(id, mass, radius, x, y, speedX, speedY, 0.0D, 0.0D);
        this.ownerHockeyistId = ownerHockeyistId;
        this.ownerPlayerId = ownerPlayerId;
    }

    /**
     * @return Возвращает идентификатор хоккеиста, контролирующего шайбу, или {@code -1}.
     */
    public long getOwnerHockeyistId() {
        return ownerHockeyistId;
    }

    /**
     * @return Возвращает идентификатор игрока, контролирующего шайбу, или {@code -1}.
     */
    public long getOwnerPlayerId() {
        return ownerPlayerId;
    }
}
